libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1244,63 +1412,74 @@ uint16_t x, usDataLength, usQuestions;
 				}
 			}
 		}
+		else
+		{
+			/* Not an expected reply. */
+		}
 #endif /* ipconfigUSE_LLMNR == 1 */
-	} while( 0 );
+		( void ) uxBytesRead;
+	} while( ipFALSE_BOOL );	/*lint !e506 !e9036 !e717 (Info -- do ... while(0);) */
 
 	if( xExpected == pdFALSE )
 	{
 		/* Do not return a valid IP-address in case the reply was not expected. */
-		ulIPAddress = 0uL;
+		ulIPAddress = 0UL;
 	}
+	#if( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS == 1 )
+	( void ) xDoStore;
+	#endif
 
 	return ulIPAddress;
 }
 /*-----------------------------------------------------------*/
 
 #if( ipconfigUSE_NBNS == 1 )
 
-	static void prvTreatNBNS( uint8_t *pucUDPPayloadBuffer,
+	static void prvTreatNBNS( uint8_t *pucPayload,
 							  size_t uxBufferLength,
 							  uint32_t ulIPAddress )
 	{
 	uint16_t usFlags, usType, usClass;
 	uint8_t *pucSource, *pucTarget;
 	uint8_t ucByte;
 	uint8_t ucNBNSName[ 17 ];
+	uint8_t *pucUDPPayloadBuffer = pucPayload;
+	NetworkBufferDescriptor_t *pxNetworkBuffer;
+	size_t uxBytesNeeded = sizeof( UDPPacket_t ) + sizeof( NBNSRequest_t );
 
 		/* Check for minimum buffer size. */
-		if( uxBufferLength < sizeof( NBNSRequest_t ) )
+		if( uxBufferLength < uxBytesNeeded )
 		{
-			return;
+			return;	/*lint !e904: Return statement before end of function [MISRA 2012 Rule 15.5, advisory]. */
 		}
 
 		/* Read the request flags in host endianness. */
-		usFlags = usChar2u16( pucUDPPayloadBuffer + offsetof( NBNSRequest_t, usFlags ) );
+		usFlags = usChar2u16( &( pucUDPPayloadBuffer[ offsetof( NBNSRequest_t, usFlags ) ] ) );
 
 		if( ( usFlags & dnsNBNS_FLAGS_OPCODE_MASK ) == dnsNBNS_FLAGS_OPCODE_QUERY )
 		{
-			usType  = usChar2u16( pucUDPPayloadBuffer + offsetof( NBNSRequest_t, usType ) );
-			usClass = usChar2u16( pucUDPPayloadBuffer + offsetof( NBNSRequest_t, usClass ) );
+			usType  = usChar2u16( &( pucUDPPayloadBuffer[ offsetof( NBNSRequest_t, usType ) ] ) );
+			usClass = usChar2u16( &( pucUDPPayloadBuffer[ offsetof( NBNSRequest_t, usClass ) ] ) );
 
 			/* Not used for now */
 			( void ) usClass;
 
 			/* For NBNS a name is 16 bytes long, written with capitals only.
 			Make sure that the copy is terminated with a zero. */
-			pucTarget = ucNBNSName + sizeof( ucNBNSName ) - 2;
-			pucTarget[ 1 ] = '\0';
+			pucTarget = &( ucNBNSName[ sizeof( ucNBNSName ) - 2U ] );
+			pucTarget[ 1 ] = ( uint8_t ) 0U;
 
 			/* Start with decoding the last 2 bytes. */
-			pucSource = pucUDPPayloadBuffer + ( offsetof( NBNSRequest_t, ucName ) + ( dnsNBNS_ENCODED_NAME_LENGTH - 2 ) );
+			pucSource = &( pucUDPPayloadBuffer[ ( dnsNBNS_ENCODED_NAME_LENGTH - 2 ) + offsetof( NBNSRequest_t, ucName ) ] );
 
 			for( ;; )
 			{
-				ucByte = ( uint8_t ) ( ( ( pucSource[ 0 ] - 0x41 ) << 4 ) | ( pucSource[ 1 ] - 0x41 ) );
+				ucByte = ( ( uint8_t ) ( ( pucSource[ 0 ] - 0x41U ) << 4 ) ) | ( pucSource[ 1 ] - 0x41U );
```

Comment:
Not blocking.
Can we make `0x41U` something more descriptive? Something like `#define ASCII_a  0x41U`.
In my opinion, the end user reading the code shouldn't need to go and look up a value. (I did)

`0U` seems okay to me since it is widely understood as Null character... But other values should be defined.

Reply:
Thanks